homeassistant/components/lupusec/config_flow.py

```diff
@@ -0,0 +1,101 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class LupusecConfigFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Lupusec config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
+
+            try:
+                await test_host_connection(self.hass, host, username, password)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if not errors:
+                await self.async_set_unique_id(host)
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=host,
+                    data={
+                        CONF_HOST: host,
+                        CONF_USERNAME: username,
+                        CONF_PASSWORD: password,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> FlowResult:
+        """Import the yaml config."""
+        self._async_abort_entries_match(user_input)
+        host = user_input[CONF_HOST]
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        try:
+            await test_host_connection(self.hass, host, username, password)
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            return self.async_abort(reason="unknown")
+
+        await self.async_set_unique_id(host)
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(title=host, data=user_input)
+
+
+async def test_host_connection(
+    hass: HomeAssistant, host: str, username: str, password: str
+):
+    """Test if the host is reachable and is actually a Lupusec device."""
+
+    try:
+        await hass.async_add_executor_job(lupupy.Lupusec, username, password, host)
+    except lupupy.LupusecException:
+        _LOGGER.error("Failed to connect to Lupusec device at %s", host)
+        raise CannotConnect
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.error("Failed to connect to Lupusec device at %s", host)
+        raise CannotConnect
```

Comment:
You're catching exception in the config flow itself to raise the "unknown" error, but you're catching it here to raise CannotConnect, so this would always be the message. I would remove this except clause.

Reply:
Removed.